narwhals/_pandas_like/dataframe.py

```diff
@@ -251,7 +253,12 @@ def _with_version(self, version: Version) -> Self:
 
     def _with_native(self, df: Any, *, validate_column_names: bool = True) -> Self:
         return self.__class__(
-            df,
+            rename_axis(
+                df,
+                implementation=self._implementation,
+                backend_version=self._backend_version,
+                columns=self._native_columns_name,
+            ),
```

Comment:
🤔 i'm wondering if we could/should just set `df.columns.name = self._native_columns_name`

else we need to trust pandas' copy-on-write / `copy=False` working properly, which i'm not sure i do

Reply:
I trust your judgement here - but should I would be concerned as a pandas user?